### PR TITLE
 refractor code and adding dynamic log level

### PR DESCRIPTION
**Pull request by banceli-dorcas**
Fixes #<issue_number_goes_here>

> It's a good idea to open an issue first for discussion.

- [ ] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: go-app/main.go [hunk 1]
```diff
@@ -106,46 +102,73 @@ func homeHandler(w http.ResponseWriter, r *http.Request) {
...
+	getFeedBack(err, "Delete Config")
+	if err == nil {
+		cloudConfigs = configs
+	}
+	feedbackTempl.Execute(w, feedBack)
+}
+func loadAnalyseLog(w http.ResponseWriter, r *http.Request) {
+	err := report.AnalyseLog(w, r, project_id, region_id)
+	if err != nil {
+		getFeedBack(err, "Delete Config")
```
Comment (msobaszek) [inline go-app/main.go:169]: Is the "Delete Config" here a typo?

File: go-app/main.go [hunk 2]
```diff
@@ -106,46 +102,73 @@ func homeHandler(w http.ResponseWriter, r *http.Request) {
...
-		}
-		feedbackTempl.Execute(w, feedBack)
+		loadDeleteConfig(w, r)
 	default:
-		err := report.AnalyseLog(w, r, project_id, region_id)
+		loadAnalyseLog(w, r)
+	}
+
+}
+func fillUploadCfgPage(w http.ResponseWriter, r *http.Request) {
```
Comment (msobaszek) [inline go-app/main.go:119]: nit: The other functions for opening a different page start with load*, can we follow the same naming convention?

File: go-app/report/analyzeLog.go [hunk 3]
```diff
@@ -160,7 +162,8 @@ func getIssueDetails(cfgFile *Config, fContent string, headerMap map[string]bool
 						if err != nil {
 							continue
 						}
-						proc_issue = strings.Join(proc_rgx_comp.FindAllString(fContent, -1), "\n")
+						raw_proc_issue := proc_rgx_comp.FindAllString(fContent, -1)
+						proc_issue = strings.Join(raw_proc_issue, "\n")
```
Comment (msobaszek) [inline go-app/report/analyzeLog.go:166]: The change here is a little unnecessary because we don't need raw_proc_issue for anything else.

**Review by msobaszek — APPROVED** [review]

File: go-app/report/analyzeLog.go (hunk 3 above)
Comment (banceli-dorcas) [reply go-app/report/analyzeLog.go:166]: I made this change because last time you said that the line is too long

File: go-app/main.go (hunk 1 above)
Comment (banceli-dorcas) [reply go-app/main.go:169]: What do you mean by that ?

File: go-app/main.go (hunk 2 above)
Comment (banceli-dorcas) [reply go-app/main.go:119]: Yes sure

**Pull request merged**